### PR TITLE
[SceneKit] Add Support for DotNet Attributes

### PR DESCRIPTION
**Pull request by tj-devel709**

### CODE REVIEW

**Review by spouliot — APPROVED** [review]

File: src/SceneKit/SCNCompat.cs [hunk 1]
```diff
@@ -52,16 +53,32 @@ public virtual void SetTimingFunction (Action<float> timingFunction)
 	}
 #if TVOS && !XAMCORE_4_0
 	partial class SCNMaterialProperty {
+#if !NET
 	[iOS (8, 0)]
 		[Deprecated (PlatformName.iOS, 10, 0)]
 		[Deprecated (PlatformName.TvOS, 10, 0, message: "This API has been totally removed on tvOS.")]
+#else
+		[UnsupportedOSPlatform ("ios10.10")]
```
Comment (mandel-macaque) [inline src/SceneKit/SCNCompat.cs:61]: No Obsolete attribute on iOS?

File: src/SceneKit/SCNCompat.cs [hunk 2]
```diff
@@ -52,16 +53,32 @@ public virtual void SetTimingFunction (Action<float> timingFunction)
...
 	partial class SCNRenderer {
+#if !NET
 		[iOS (8, 0)]
 		[Deprecated (PlatformName.iOS, 9, 0)]
 		[Deprecated (PlatformName.TvOS, 10, 0, message: "This API has been totally removed on tvOS.")]
+#else
+		[UnsupportedOSPlatform ("ios10.10")]
+		[UnsupportedOSPlatform ("tvos10.10")]
+#if TVOS
+		[Obsolete ("Starting with tvos10.10 this API has been totally removed on tvOS.", DiagnosticId = "BI1234", UrlFormat = "https://github.com/xamarin/xamarin-macios/wiki/Obsolete")]
```
Comment (mandel-macaque) [inline src/SceneKit/SCNCompat.cs:79]: same, no iOS Obsolete.

**Review by mandel-macaque — CHANGES_REQUESTED** [review]

File: src/SceneKit/SCNCompat.cs (hunk 2 above)
Comment (tj-devel709) [reply src/SceneKit/SCNCompat.cs:79]: There was only a message associated with the the TvOS deprecation and there doesn't seem to be more info on the web docs about the deprecation message. Should I add the same message for iOS and just say totally removed on iOS?

File: src/SceneKit/SCNCompat.cs (hunk 1 above)
Comment (tj-devel709) [reply src/SceneKit/SCNCompat.cs:61]: Same with this selector!

Comment (mandel-macaque) [reply src/SceneKit/SCNCompat.cs:61]: Looking at the headers, the same message applies.

File: src/SceneKit/SCNCompat.cs (hunk 2 above)
Comment (mandel-macaque) [reply src/SceneKit/SCNCompat.cs:79]: Same, looking at the headers, it was completely removed.

File: src/SceneKit/SCNCompat.cs [hunk 3]
```diff
@@ -55,13 +55,15 @@ public virtual void SetTimingFunction (Action<float> timingFunction)
 	partial class SCNMaterialProperty {
 #if !NET
 	[iOS (8, 0)]
-		[Deprecated (PlatformName.iOS, 10, 0)]
+		[Deprecated (PlatformName.iOS, 10, 0, message: "This API has been totally removed on iOS.")]
 		[Deprecated (PlatformName.TvOS, 10, 0, message: "This API has been totally removed on tvOS.")]
 #else
 		[UnsupportedOSPlatform ("ios10.10")]
```
Comment (mandel-macaque) [inline src/SceneKit/SCNCompat.cs:61]: 10.10 is not a version.

File: src/SceneKit/SCNCompat.cs [hunk 4]
```diff
@@ -70,13 +72,15 @@ partial class SCNMaterialProperty {
 	partial class SCNRenderer {
 #if !NET
 		[iOS (8, 0)]
-		[Deprecated (PlatformName.iOS, 9, 0)]
+		[Deprecated (PlatformName.iOS, 9, 0, message: "This API has been totally removed on iOS.")]
 		[Deprecated (PlatformName.TvOS, 10, 0, message: "This API has been totally removed on tvOS.")]
 #else
-		[UnsupportedOSPlatform ("ios10.10")]
+		[UnsupportedOSPlatform ("ios9.0")]
 		[UnsupportedOSPlatform ("tvos10.10")]
```
Comment (mandel-macaque) [inline src/SceneKit/SCNCompat.cs:79]: 10.10 is not a version.

**Review by mandel-macaque — CHANGES_REQUESTED** [review]

**Review by mandel-macaque — APPROVED** [review]

**Pull request merged**